packages/kbn-reporting/public/share/shared/get_shared_components.tsx

```diff
@@ -7,23 +7,22 @@
  */
 
 import { CoreSetup } from '@kbn/core/public';
-import React from 'react';
-
-import { PDF_REPORT_TYPE, PDF_REPORT_TYPE_V2 } from '@kbn/reporting-export-types-pdf-common';
+import { JobParamsCSV } from '@kbn/reporting-export-types-csv-common';
+import { JobAppParamsPDFV2, PDF_REPORT_TYPE_V2 } from '@kbn/reporting-export-types-pdf-common';
 import { PNG_REPORT_TYPE_V2 } from '@kbn/reporting-export-types-png-common';
-
-import { ReportingAPIClient } from '../../reporting_api_client';
+import React from 'react';
+import { ReportingAPIClient } from '../..';
 import { ReportingPanelProps } from '../share_context_menu/reporting_panel_content';
 import { ScreenCapturePanelContent } from '../share_context_menu/screen_capture_panel_content_lazy';
-
 /**
  * Properties for displaying a share menu with Reporting features.
  */
 export interface ApplicationProps {
   /**
    * A function that Reporting calls to get the sharing data from the application.
+   * Needed for CSV exports and Canvas PDF reports.
    */
-  getJobParams: ReportingPanelProps['getJobParams'];
+  getJobParams?: JobAppParamsPDFV2 | JobParamsCSV | ReportingPanelProps['getJobParams'];
```

Comment:
Looks like you can revert this
```suggestion
  getJobParams: ReportingPanelProps['getJobParams'];
```